flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/aggfunctions/LeadLagAggFunction.java

```diff
@@ -230,6 +230,18 @@ public DataType getResultType() {
         }
     }
 
+    /** CharLeadLagAggFunction. */
+    public static class CharLeadLagAggFunction extends LeadLagAggFunction {
+        public CharLeadLagAggFunction(int operandCount) {
+            super(operandCount);
+        }
+
+        @Override
+        public DataType getResultType() {
+            return DataTypes.CHAR(operandCount());
```

Comment:
This `operandCount()` is always 3. You might need to add another parameter in the constructor to feed in the precision of the char type and use that precision here.

Reply:
thx. I've rewritten this code like the way handle DECIMAL type fields.